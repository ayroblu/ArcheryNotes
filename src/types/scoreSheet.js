const prefix = 'SCORE_SHEET/'

export const SET = prefix + 'SET'
export const RESET = prefix + 'RESET'