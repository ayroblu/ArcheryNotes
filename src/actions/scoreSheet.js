import { SET, RESET } from '../types/scoreSheet'

export function set(payload){
  return {
    type: SET
  , payload
  }
}

export function reset(payload={}){
  return {
    type: RESET
  , payload
  }
}