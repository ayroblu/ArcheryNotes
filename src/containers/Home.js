import React, {Component} from 'react'
import {
  View,
  ScrollView,
  Text,
  StyleSheet,
  StatusBar,
} from 'react-native'
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'

import DevTools from '../components/DevTools'
import {navButtons, onNavigatorEvent} from '../utils/nav'
import * as mainActions from '../actions/main'
import * as theme from '../theme'
import {textStyle} from '../config'

class Home extends Component {
  static navigatorButtons = navButtons
  componentWillMount(){
    this.props.mainActions.set({navigator: this.props.navigator})
    this.props.navigator.setOnNavigatorEvent(onNavigatorEvent.bind(this));
  }
  _renderRow(row, rowId){
    const isFirstLine = row.preTotal === undefined
    const rowTotal = row.scores.filter(s=>s).reduce((a,n)=>a+n,0)
    return (
      <View style={styles.line} key={rowId}>
        {row.scores.map((r,i)=>(
          <Text style={styles.boxText} key={i}>
            {r}
          </Text>
        ))}
        <Text style={styles.subTotalBoxText}>
          {rowTotal}
        </Text>
        <Text style={isFirstLine ? styles.totalBox : styles.totalBoxText}>
          {!isFirstLine && (row.preTotal + rowTotal)}
        </Text>
      </View>
    )
  }
  render() {
    const {text} = this.props.main
    const {users} = this.props.scoreSheet
    const rows = users[0].scoreSheet
    return (
      <ScrollView style={styles.flex} contentContainerStyle={styles.container}>
        <StatusBar barStyle="light-content" />
        <Text style={styles.text}>{text}</Text>
        <Text style={styles.text}>Archery Notes</Text>
        <Text style={styles.text}>Probably should add a score card here</Text>
        <View style={styles.scoreSheet}>
          {rows.map((r,i)=>(
            this._renderRow(r,i)
          ))}
        </View>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center'
  }
, flex: {
    flex: 1
  }
, text: {
    ...textStyle
  }
, scoreSheet: {
    marginTop: 10
  , alignSelf: 'stretch'
  }
, line: {
    flexDirection: 'row'
  , marginBottom: 4
  }
, boxText: {
    ...textStyle
  , flex: 1
  , textAlign: 'center'
  , backgroundColor: '#ddd'
  , marginLeft: 4
  }
, totalBox: {
    marginLeft: 4
  , marginRight: 4
  , flex: 2
  }
})
styles.totalBoxText = StyleSheet.flatten([styles.boxText, styles.totalBox])
styles.subTotalBoxText = StyleSheet.flatten([styles.boxText, {flex: 1.5}])

export default connect(state=>({
  main: state.main
, scoreSheet: state.scoreSheet
}), dispatch=>({
  mainActions: bindActionCreators(mainActions, dispatch)
}))(Home)
