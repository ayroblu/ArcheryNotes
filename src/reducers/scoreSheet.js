import { SET, RESET } from '../types/scoreSheet'

const initialState = {
  users: [{
    name: 'first'
  , scoreSheet: [{
      scores: [null, null, null, null, null, null]
    }]
  }]
}

export default function reducer(state=initialState, action) {
  switch (action.type) {
    case SET:
      return {...state, ...action.payload}
    case RESET:
      return {...initialState, ...action.payload}
    default:
      return state
  }
}
