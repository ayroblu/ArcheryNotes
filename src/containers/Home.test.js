import 'react-native'
import React from 'react'
import {Provider} from 'react-redux'
import Home from './Home'
import configureStore from '../store'
import * as ssActions from '../actions/scoreSheet'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const store = configureStore()

it('renders correctly', async ()=>{
  const props = {
    navigator: {
      setOnNavigatorEvent(){}
    }
  }
  store.dispatch(ssActions.set({
    users: [{
      name: 'first'
    , scoreSheet: [{
        scores: [9,8,7,6,5,4]
      }, {
        scores: [9,8,7,6,5,null]
      , preTotal: 39
      }]
    }]
  }))
  await new Promise(y=>setTimeout(y))
  const tree = renderer.create(
    <Provider store={store}>
      <Home {...props} />
    </Provider>
  )
  expect(tree).toMatchSnapshot()
})
