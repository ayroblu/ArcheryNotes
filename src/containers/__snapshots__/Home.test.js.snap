// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#555",
          "fontSize": 18,
          "fontWeight": "300",
        }
      }
    >
      thing
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#555",
          "fontSize": 18,
          "fontWeight": "300",
        }
      }
    >
      Archery Notes
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#555",
          "fontSize": 18,
          "fontWeight": "300",
        }
      }
    >
      Probably should add a score card here
    </Text>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 4,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          9
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          8
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          7
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          6
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          5
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          4
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1.5,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          39
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 2,
              "marginLeft": 4,
              "marginRight": 4,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 4,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          9
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          8
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          7
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          6
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          5
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 1.5,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "textAlign": "center",
            }
          }
        >
          35
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "#ddd",
              "color": "#555",
              "flex": 2,
              "fontSize": 18,
              "fontWeight": "300",
              "marginLeft": 4,
              "marginRight": 4,
              "textAlign": "center",
            }
          }
        >
          74
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
